tests/blackbox/src/validation/utils/models.py

```diff
@@ -0,0 +1,91 @@
+import os
+from typing import Protocol
+
+import yaml
+from common.logger import get_logger
+from gen_ai_hub.proxy import get_proxy_client
+from gen_ai_hub.proxy.langchain import ChatOpenAI
+from gen_ai_hub.proxy.native.google.clients import GenerativeModel
+from pydantic import BaseModel
+
+logger = get_logger(__name__)
+
+proxy_client = get_proxy_client("gen-ai-hub")
+
+
+class ModelConfig(BaseModel):
+    name: str
+    deployment_id: str
+
+
+class Model(Protocol):
+    def invoke(self, content: str) -> str: ...
+
+    @property
+    def name(self) -> str: ...
+
+
+class OpenAIModel(Model):
```

Comment:
Shouldn't either this or `Model` itself also inherit from `BaseModel`? Otherwise we will not get Pydantic in here.

Reply:
The `Model` interface is our own to be used to wrap openai, gemini, claude, etc. models.